### PR TITLE
webAppGitHub

### PR DESCRIPTION
**Pull request by Krisnatagoras**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 201-web-app-github-deploy/README.md [hunk 1]
```diff
@@ -9,14 +9,28 @@
...
+<img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.png"/>
+</a><a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-web-app-github-deploy%2Fazuredeploy.json" target="_blank">
+<img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.png"/>
 </a>
 
-For information about using this template, see [Deploy a web app linked to a GitHub repository](https://azure.microsoft.com/en-us/documentation/articles/app-service-web-arm-from-github-provision/).
+This template deploys a **simple Linux Web Application linked with GitHub Repository**. The WebPlan is settled for Consumption. For more information about hosting Plans click [here!](https://azure.microsoft.com/en-gb/pricing/details/app-service/linux/). As a default, this template deploys the web app using the "S1" (free tier) as the pricing tier for the hosting plan. It is ideal for development and test environments, for production environments you must consider different plans.
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/README.md]: ... linked to a GitHub Repository...

File: 201-web-app-github-deploy/README.md [hunk 2]
```diff
@@ -9,14 +9,28 @@
...
+
+## Prerequisites
+
+### Have a Git Hub Account
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/README.md]: ### A GitHub Account

File: 201-web-app-github-deploy/README.md [hunk 3]
```diff
@@ -9,14 +9,28 @@
...
+
+If you don't have yet a Git Hub account, you can create your account on [Git Hub](https://github.com/).
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/README.md]: GitHub

File: 201-web-app-github-deploy/README.md [hunk 4]
```diff
@@ -9,14 +9,28 @@
...
+
+If you are new to **Web App Service**, see:
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/README.md]: Services

File: 201-web-app-github-deploy/azuredeploy.json [hunk 5]
```diff
@@ -1,114 +1,113 @@
...
-      "defaultValue": "master",
-      "metadata": {
-        "description": "The branch of the GitHub repository to use."
-      }
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.1",
+    "parameters": {
+        "webAppName": {
+           "type": "string",
+           "defaultValue": "[concat('WebApp-', uniqueString(resourceGroup().name, utcNow('F')))]",
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/azuredeploy.json]: simplify this and make it idempotent

"defaultValue": "[concat('WebApp-', uniqueString(resourceGroup().id))]",

File: 201-web-app-github-deploy/azuredeploy.json [hunk 6]
```diff
@@ -1,114 +1,113 @@
...
-      "location": "[parameters('location')]",
-      "sku": {
-        "name": "[parameters('sku')]",
-        "capacity": "[parameters('workerSize')]"
-      },
-      "properties": {
-        "name": "[parameters('hostingPlanName')]"
-      }
+    "variables": {
+        "hostingPlanName": "[concat(resourceGroup().name, '-hpn')]"
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/azuredeploy.json]: Put the constant first in the name, eg

"hostingPlanName": "[concat('hpn-', resourceGroup().name)]"

File: 201-web-app-github-deploy/azuredeploy.json [hunk 7]
```diff
@@ -1,114 +1,113 @@
...
+                "2"
+            ],
+            "defaultValue": "0",
+            "metadata": {
+                "description": "The instance size of the hosting plan (small, medium, or large)."
+            }
+        },
+        "repoURL": {
+            "type": "string",
+            "defaultValue": "https://github.com/Azure/azure-quickstart-templates.git",
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/azuredeploy.json]: this isn't going to actually do anything on the web app... we need to have some sample code/repo that we can pull from here...

File: 201-web-app-github-deploy/metadata.json [hunk 8]
```diff
@@ -1,11 +1,17 @@
...
-  "itemDisplayName": "Deploy a web app linked to a GitHub repository",
-  "description": "This template creates a web app that is linked to a project in a GitHub repository.",
-  "summary": "Deploy a web app linked to a GitHub repository",
-  "githubUsername": "cephalin",
-  "dateUpdated": "2016-04-25"
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "itemDisplayName": "WebApp GitHub",
+    "description": "This template allows you to create an WebApp linked with a GitHub Repository linked.",
+    "summary": "Deploy a WebApp Service  liked with a GitHub Repository.",
+    "githubUsername": "Krisnatagoras",
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/metadata.json]: restore the original author's name here

File: 201-web-app-github-deploy/README.md (hunk 1 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/README.md]: Thanks for the feedback Brian.
All changes on the Readme are done.

File: 201-web-app-github-deploy/README.md (hunk 2 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/README.md]: Done!

File: 201-web-app-github-deploy/README.md (hunk 3 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/README.md]: Sure thing!

File: 201-web-app-github-deploy/README.md (hunk 4 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/README.md]: Done.

File: 201-web-app-github-deploy/azuredeploy.json (hunk 5 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/azuredeploy.json]: Makes more sense... also done!

File: 201-web-app-github-deploy/azuredeploy.json (hunk 6 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/azuredeploy.json]: Done!

File: 201-web-app-github-deploy/metadata.json (hunk 8 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/metadata.json]: Cool.

File: 201-web-app-github-deploy/azuredeploy.json (hunk 7 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/azuredeploy.json]: So, my difficult here is if I let the parameter empty on the checks it complains that it's a required parameter. Do you have any tip on how should I fix this?

Comment (leestott) [reply 201-web-app-github-deploy/azuredeploy.json]: @Krisnatagoras use this repo https://github.com/Microsoft/project-html-website

Comment (Krisnatagoras) [reply 201-web-app-github-deploy/azuredeploy.json]: Sure thing @leestott. Done!

File: 201-web-app-github-deploy/azuredeploy.json [hunk 9]
```diff
@@ -2,9 +2,9 @@
     "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
     "contentVersion": "1.0.0.1",
     "parameters": {
-        "webAppName": {
+        "siteName": {
            "type": "string",
-           "defaultValue": "[concat('WebApp-', uniqueString(resourceGroup().name, utcNow('F')))]",
+           "defaultValue": "[concat('WebApp-', uniqueString(resourceGroup().name))]",
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/azuredeploy.json]: use resourceGroup().id (not .name)

File: 201-web-app-github-deploy/azuredeploy.json [hunk 10]
```diff
@@ -51,7 +51,7 @@
         },
         "repoURL": {
             "type": "string",
-            "defaultValue": "https://github.com/Azure/azure-quickstart-templates.git",
+            "defaultValue": "https://github.com/Microsoft/project-html-website",
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/azuredeploy.json]: the readme for this repo says the repo is no longer used?  Do we have an Azure4Students account or repo we could link to?

Comment (Krisnatagoras) [reply 201-web-app-github-deploy/azuredeploy.json]: As @leestott  recommended, I'm using this repo: https://github.com/Microsoft/project-html-website

File: 201-web-app-github-deploy/azuredeploy.json (hunk 9 above)
Comment (Krisnatagoras) [reply 201-web-app-github-deploy/azuredeploy.json]: Changed.

File: 201-web-app-github-deploy/azuredeploy.json [hunk 11]
```diff
@@ -1,114 +1,113 @@
...
-      }
-    },
-    "branch": {
-      "type": "string",
-      "defaultValue": "master",
-      "metadata": {
-        "description": "The branch of the GitHub repository to use."
-      }
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.1",
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/azuredeploy.json]: revert this change

File: 201-web-app-github-deploy/azuredeploy.json (hunk 10 above)
Comment (bmoore-msft) [reply 201-web-app-github-deploy/azuredeploy.json]: @leestott can we find a supported repo for this sample?

File: 201-web-app-github-deploy/azuredeploy.json (hunk 11 above)
Comment (leestott) [reply 201-web-app-github-deploy/azuredeploy.json]: Hi So I have created this repo @Krisnatagoras @bmoore-msft   for deploying a test website https://github.com/leestott/WebSiteTest.git

File: 201-web-app-github-deploy/azuredeploy.json [hunk 12]
```diff
@@ -51,7 +51,7 @@
         },
         "repoURL": {
             "type": "string",
-            "defaultValue": "https://github.com/Microsoft/project-html-website",
+            "defaultValue": "https://github.com/leestott/WebSiteTest.git",
```
Comment (bmoore-msft) [inline 201-web-app-github-deploy/azuredeploy.json]: We can't use a personal repo here - is there nothing supported in github we can use?

Comment (leestott) [reply 201-web-app-github-deploy/azuredeploy.json]: @bmoore-msft  what about https://github.com/Azure-Samples/app-service-web-html-get-started.git is this good?

Comment (Krisnatagoras) [reply 201-web-app-github-deploy/azuredeploy.json]: Changed the repo.

Comment (leestott) [reply 201-web-app-github-deploy/azuredeploy.json]: @bmoore-msft can you please check this over and merge if your happy.
